crates/stackable-telemetry/src/tracing.rs

```diff
@@ -186,15 +181,9 @@ impl Tracing {
             let otel_tracer = opentelemetry_otlp::new_pipeline()
                 .tracing()
                 .with_exporter(trace_exporter)
-                .with_trace_config(
-                    trace::config()
-                        .with_sampler(Sampler::AlwaysOn) // TODO (@NickLarsenNZ): Make this configurable. See also Sampler::ParentBased
-                        .with_id_generator(RandomIdGenerator::default())
-                        .with_resource(Resource::new(vec![KeyValue::new(
-                            resource::SERVICE_NAME,
-                            self.service_name,
-                        )])),
-                )
+                .with_trace_config(trace::config().with_resource(Resource::new(vec![
```

Comment:
question: Was the sampler not needed anymore. We discussed explicitly setting the ID generator, because it uses the random ID generator by default.

Reply:
The default ID Generator is the [RandomIdGenerator], so it seems redundant to make it explicit. The only other generate I know of is the [XrayIdGenerator] (maybe it needs to be configurable one day, but we should wait for the need).

As for the sampler, the default uses [ParentBased] and falling back to [AlwaysOn] (it allows parent spans that might be probability based to also pass that info on).

Default impl can be found here: https://github.com/open-telemetry/opentelemetry-rust/blob/b933bdd82dadbadedb42a1f572caba1e0b8b9391/opentelemetry-sdk/src/trace/config.rs#L95-L96

[RandomIdGenerator]: https://docs.rs/opentelemetry_sdk/0.22.0/opentelemetry_sdk/trace/struct.RandomIdGenerator.html
[XrayIdGenerator]: https://docs.rs/opentelemetry_sdk/0.22.0/opentelemetry_sdk/trace/struct.XrayIdGenerator.html
[ParentBased]: https://docs.rs/opentelemetry_sdk/0.22.0/opentelemetry_sdk/trace/enum.Sampler.html#variant.ParentBased
[AlwaysOn]: https://docs.rs/opentelemetry_sdk/0.22.0/opentelemetry_sdk/trace/enum.Sampler.html#variant.AlwaysOn

Reply:
Thank you for the great explanation!